src/timepicker/timepicker.js

```diff
@@ -254,9 +312,14 @@ angular.module('ui.bootstrap.timepicker', [])
     $scope.meridian = selected.getHours() < 12 ? meridians[0] : meridians[1];
   }
 
-  function addMinutes( minutes ) {
-    var dt = new Date( selected.getTime() + minutes * 60000 );
-    selected.setHours( dt.getHours(), dt.getMinutes() );
+  function addMinutes( date,  minutes ) {
+    var dt = new Date( date.getTime() + minutes * 60000 );
+    var newDate = new Date( date.getTime() ).setHours( dt.getHours(), dt.getMinutes() );
+    return new Date( newDate );
```

Comment:
Is there any reason why `newDate` can't just get returned? It should already be an independent reference.